formats/json/commonMain/src/kotlinx/serialization/json/internal/lexer/AbstractJsonLexer.kt

```diff
@@ -184,7 +184,6 @@ internal abstract class AbstractJsonLexer {
 
     open fun consumeNextToken(expected: Char) {
         ensureHaveChars()
-        val source = source
```

Comment:
It was inserted here because access to a local variable is faster than to a field getter, but we need to test with benchmarks that JVM can optimize this for us

Reply:
We cannot use a local variable here, because in the implementation in `ReaderJsonLexer` this field is mutable.
If during the call `prefetchOrEof(cpos)`the `source` field will be changed, then in this case we will read the old cached array

Reply:
You can leave it so, or make the `source` field immutable.

Reply:
Ok then

Reply:
I have added a commit with corrections, please watch it (`ReaderJsonLexer#source` now immutable)